examples/rust/full-wallet/src/main.rs

```diff
@@ -0,0 +1,102 @@
+// const DESCRIPTOR_PRIVATE_EXTERNAL: &str = "tr(tprv8ZgxMBicQKsPdJuLWWArdBsWjqDA3W5WoREnfdgKEcCQB1FMKfSoaFz9JHZU71HwXAqTsjHripkLM62kUQar14SDD8brsmhFKqVUPXGrZLc/86'/1'/0'/0/*)#fv8tutn2";
+// const DESCRIPTOR_PRIVATE_INTERNAL: &str = "tr(tprv8ZgxMBicQKsPdJuLWWArdBsWjqDA3W5WoREnfdgKEcCQB1FMKfSoaFz9JHZU71HwXAqTsjHripkLM62kUQar14SDD8brsmhFKqVUPXGrZLc/86'/1'/0'/1/*)#ccz2p7rj";
+// --8<-- [end:descriptors]
+
+fn main() -> Result<(), anyhow::Error> {
+    // --8<-- [start:persist]
+    let mut conn = Connection::open(DB_PATH)?;
+
+    let wallet_opt = Wallet::load()
+        .descriptor(KeychainKind::External, Some(DESCRIPTOR_PRIVATE_EXTERNAL))
+        .descriptor(KeychainKind::Internal, Some(DESCRIPTOR_PRIVATE_INTERNAL))
+        .extract_keys()
+        .check_network(Network::Signet)
+        .load_wallet(&mut conn)?;
+
+    let (mut wallet, is_new_wallet) = if let Some(loaded_wallet) = wallet_opt {
+        (loaded_wallet, false)
+    } else {
+        (Wallet::create(DESCRIPTOR_PRIVATE_EXTERNAL, DESCRIPTOR_PRIVATE_INTERNAL)
+            .network(Network::Signet)
+            .create_wallet(&mut conn)?, true)
+    };
+    // --8<-- [end:persist]
+
+    // --8<-- [start:scan]
+    let client: esplora_client::BlockingClient = Builder::new("https://mutinynet.com/api").build_blocking();
+    // Sync the wallet
+    if is_new_wallet {
+        // Perform a full scan
+        println!("Performing full scan...");
+        let full_scan_request: FullScanRequestBuilder<KeychainKind> = wallet.start_full_scan();
+        let update: FullScanResult<KeychainKind> = client.full_scan(full_scan_request, STOP_GAP, PARALLEL_REQUESTS)?;
+        wallet.apply_update(update).unwrap();
+    } else {
+        // Perform a regular sync
+        println!("Performing regular sync...");
+        let sync_request: SyncRequestBuilder<(KeychainKind, u32)> = wallet.start_sync_with_revealed_spks();
+        let update: SyncResult = client.sync(sync_request, PARALLEL_REQUESTS)?;
+        wallet.apply_update(update).unwrap();
+    };
+    wallet.persist(&mut conn)?;
+    // --8<-- [end:scan]
+
+    // --8<-- [start:address]
+    // Reveal a new address from your external keychain
+    let address: AddressInfo = wallet.reveal_next_address(KeychainKind::External);
+    println!("Generated address {} at index {}", address.address, address.index);
+    wallet.persist(&mut conn)?;
+    // --8<-- [end:address]
+
+    let balance = wallet.balance();
+    println!("Wallet balance: {} sat", balance.total().to_sat());
+
+    // --8<-- [start:faucet]
+    // Use the Mutinynet faucet return address
+    let address = Address::from_str("tb1qd28npep0s8frcm3y7dxqajkcy2m40eysplyr9v")
+        .unwrap()
+        .require_network(Network::Signet)
+        .unwrap();
+
+    let send_amount: Amount = Amount::from_sat(5000);
+    // return all sats to mutinynet faucet
+    // let send_amount: Amount = Amount::from_sat(balance.total().to_sat() - 100);// min fee is 100 sat
```

Comment:
hmm, well I tried sending all and the error said it required 100, tried that and it worked. my understanding was that mutiny net has a hard coded min fee of 100, but that could totally be wrong, it could be variable. maybe for now I'll just remove this and the note about it and make a separate PR for this soon (as you suggest below)

Reply:
Ah good to know! I didn't know that mutinynet was using a hard-coded minimum fee. In this case it's non-standard and odd so I definitely wouldn't want to use that and make an example of it. 

Note also that MutinyNet will eventually be sunset (just a hunch) and so I don't see the examples sticking to MutinyNet for the long run (but it works now and I don't want to slow down the PRs so I'm cool with using it).

Reply:
there has been some talk of a bdk hosted signet, but for now mutiny works

Reply:
Yeah we'll move to full standard Signet if we can find a good and reliable faucet, or to testnet once the Testnet4 stabilizes.